### DISCUSSION TITLE
Use of PolycrystalVariableAction-created variables in inputfiles

### DISCUSSION THREAD

**Discussion by leanderVS** [idaholab/moose#16871] [Q&A Modules: Phase field] [answered]
Hi all

I recently started looking into the possibilities of MOOSE for corrosion modelling using the Phase Field module. 

One of my very simple ideas would be to adapt the PolycrystalKernelAction. For this I simple assume the electrolyte to be another order parameter within the PolycrystalVariablesAction system, i.e. gr_1/eta_1 refers to the electrolyte. To limit corrosion to the interface, I adapted the GBEvolutionBase.C file, such that L~gr_1 (very simple, but theoretically effective for a first try). So basically I am slightly adapting the available grain_growth_2D_random.i file within the phase field module.

On the long term, I would use EBSD input and stitch that input to an electrolyte mesh. For now, however, I am just manually defining two grains and one electrolyte phase (through the FunctionIC system), while still using the abovementioned actions for variable/kernel generation. The thing is, when I try using gr_1 within the inputfile to supply as input for the FunctionIC or GBEvolutionBase.C adaptation, it just doesn't recognize it.

*** ERROR ***      Variable 'gr_1' requested in initial condition 'eta_electrolyte' does not exist. 

Which makes me wonder if it is somehow related to the order of the blocks within the inputfile or if you just can't refer to internally/Action-generated variables? Well, I am new, so I might be making a rookie mistake here. 

Kind regards

Leander Van Speybroeck
Ghent University, Belgium

**Comment by SudiptaBiswas** [comment, marked as answer]
The variable name that comes out of the `PolycrystalVariableAction` are `gr0 gr1 ..` or `eta0 eta1 ..` etc., based on the parameter `var_name_base`. Make sure you are proving the right variable name in your ICs and Materials. To avoid confusion, it might be better to manually add a separate variable in the input file for the electrolyte.

**Reply by leanderVS** [reply]
Thank you for your reply. Turns out I just kept mixing up the PolycrystalVariableAction source code. Indeed, the generated variables are called gr0, gr1, gr2 instead of gr_1, gr_2, gr_3. Thanks again for the quick fix!
My last resort was indeed to implement the (electrolyte) order parameter separately.
